geopandas/tests/test_sindex.py

```diff
@@ -43,6 +55,27 @@ def test_no_sindex_active(self):
 @pytest.mark.skipif(sys.platform.startswith("win"), reason="fails on AppVeyor")
 @pytest.mark.skipif(not sindex.has_sindex(), reason="Spatial index absent, skipping")
 class TestSeriesSindex:
+    def test_sindex_generated(self):
+        """Test the sindex_generated method."""
+        t1 = Polygon([(0, 0), (1, 0), (1, 1)])
+        t2 = Polygon([(0, 0), (1, 1), (0, 1)])
+        s = GeoSeries([t1, t2])
+        d = GeoDataFrame({"location": [t1, t2]}, geometry="location")
+        assert not d.sindex_generated()
```

Comment:
Recommend separating the tests of the GeoDataFrame from the tests of the GeoSeries.  These can stay in this function but should be 2 separate blocks, .e.g,
```
assert not s.index_generated()
s.sindex
assert s.index_generated()
```

Reply:
Yup implemented